### ISSUE TITLE
LMMS wants to remove wine-staging on Ubuntu

### ISSUE THREAD

**Issue by mypickedusername** [LMMS/lmms#3705]
When I try to install LMMS from Ubuntu's repo it wants to remove wine-staging that I have installed on my system and install regular wine instead. I don't want it to remove wine-staging. What can I do to fix this?

**Comment by tresf** [comment]
@mypickedusername this is really a question for Ubuntu, but I'm guessing they packaged lmms against a specific Wine version.  There's not much you can do in this scenario, you'll have to pick one or the other.

We're working on some portable images of LMMS (`.AppImage`) which would allow you to move away from the package manager in general, if interested.  More info here: #3688

**Comment by karmux** [comment]
One option is to compile LMMS from source. This way it can use wine-staging or wine-devel if corresponding dev packages are installed.

I think that Ubuntu binary package should not install wine-stable if either wine-devel or wine-staging is already installed. But as @tresf said it's Ubuntu's issue.

**Comment by mypickedusername** [comment]
Compiling from source didn't work for me either because when I tried to install the required dependencies there was exactly the same issue. Some of them wanted me to get rid of wine-staging.

Does that mean that I currently can't install/compile LMMS on Ubuntu and keep using wine-staging? Do you think there could be some way to force the package manager to not install wine?

**Comment by mypickedusername** [comment]
@tresf that app image works very well, thanks!

**Comment by karmux** [comment]
I'm using compiled LMMS with wine-staging on Ubuntu.
You need to install following Wine packages from WineHQ repos:
```
$ dpkg -l | grep wine | awk '{print $2}'
wine-staging
wine-staging-amd64
wine-staging-dev
wine-staging-i386:i386
winehq-staging
```

**Comment by karmux** [comment]
@mypickedusername did you install gcc-multilib and g++-multilib?

**Comment by mypickedusername** [comment]
Yes I did after I saw those warnings and nothing changed. The same warnings still appear.

**Comment by PhysSong** [comment]
`stable-1.2` doesn't seems to support wineHQ, while `master` does.
@LMMS/developers Could we backport #3278 and #3708?

**Comment by karmux** [comment]
@PhysSong Wine-Staging and Wine-Devel for `stable-1.2` is in #3688.

**Comment by PhysSong** [comment]
@karmux Thanks! I didn't konw that.

**Comment by tresf** [comment]
@PhysSong sorry for the delay merging #3688.  I was waiting on @lukas-w to help out with this comment: https://github.com/LMMS/lmms/pull/3688#issuecomment-333726469.  We can merge now and open a new issue if this makes more sense.

**Comment by zonkmachine** [comment]
> We can merge now and open a new issue if this makes more sense.

It makes enough sense. That is one long issue and it's getting hard to navigate.

**Comment by tresf** [comment]
#3688 has been merged.  The Ubuntu packaging issue isn't our problem and the OP states the AppImage fixes the linking issues.  Closing.
